Mathlib/NumberTheory/Rayleigh.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2023 Jason Yuen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jason Yuen
+-/
+import Mathlib.Data.Real.ConjugateExponents
+import Mathlib.Data.Real.Irrational
+
+/-!
+# Rayleigh's theorem on Beatty sequences
+
+This file proves Rayleigh's theorem on Beatty sequences. We start by proving `rayleigh_compl`, which
+is a generalization of Rayleigh's theorem, and eventually prove `rayleigh_irr_pos`, which is
+Rayleigh's theorem.
+
+## Main definitions
+
+* `beattySequence`: In the Beatty sequence for real number `r`, the `k`th term is `⌊k * r⌋`.
+* `beattySequence'`: In this variant of the Beatty sequence for `r`, the `k`th term is
+  `⌈k * r⌉ - 1`.
+
+## Main statements
+
+Define the following Beatty sets, where `r` denotes a real number:
+
+* `B_r := {⌊k * r⌋ | k ∈ ℤ}`
+* `B'_r := {⌈k * r⌉ - 1 | k ∈ ℤ}`
+* `B⁺_r := {⌊r⌋, ⌊2r⌋, ⌊3r⌋, ...}`
+* `B⁺'_r := {⌈r⌉-1, ⌈2r⌉-1, ⌈3r⌉-1, ...}`
+
+The main statements are:
+
+* `compl_beattySequence`: Let `r` be a real number greater than 1, and `1/r + 1/s = 1`.
+  Then the complement of `B_r` is `B'_s`.
+* `beattySequence_symmDiff_beattySequence'_pos`: Let `r` be a real number greater than 1, and
+  `1/r + 1/s = 1`. Then `B⁺_r` and `B⁺'_s` partition the positive integers.
+* `Irrational.beattySequence_symmDiff_beattySequence_pos`: Let `r` be an irrational number greater
+  than 1, and `1/r + 1/s = 1`. Then `B⁺_r` and `B⁺_s` partition the positive integers.
+
+## References
+
+* [Wikipedia, *Beatty sequence*](https://en.wikipedia.org/wiki/Beatty_sequence)
+
+## Tags
+
+beatty, sequence, rayleigh, irrational, floor, positive
+-/
+
+/-- In the Beatty sequence for real number `r`, the `k`th term is `⌊k * r⌋`. -/
+noncomputable def beattySequence (r : ℝ) : ℤ → ℤ :=
+  fun k ↦ ⌊k * r⌋
+
+/-- In this variant of the Beatty sequence for `r`, the `k`th term is `⌈k * r⌉ - 1`. -/
+noncomputable def beattySequence' (r : ℝ) : ℤ → ℤ :=
+  fun k ↦ ⌈k * r⌉ - 1
+
+namespace Beatty
+
+variable {r s : ℝ} (hrs : r.IsConjugateExponent s) {j k : ℤ}
+
+/-- Let `r > 1` and `1/r + 1/s = 1`. Then `B_r` and `B'_s` are disjoint (i.e. no collision exists).
+-/
+theorem no_collision : Disjoint {beattySequence r k | k} {beattySequence' s k | k} := by
```

Comment:
I would advise to make this lemma and the next two private.

Reply:
Done. The helper theorems are all private now.